web/ui/react-app/src/api/FetchWithStatus.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { FC, ComponentType } from 'react';
+import { Fetch, FetchState } from '../api/Fetch';
+import { StatusIndicator } from '../StatusIndicator';
+import PathPrefixProps from '../PathPrefixProps';
+
+export const fetchWithStatus = <T extends {}, DataType extends {}>(
```

Comment:
Hm, why not `fetchWithStatus` -> `FetchWithStatus` like other components?

...and I'm still trying to understand the crazy TypeScript signature :laughing:

Reply:
Ah ok, because it's a HOC, which explains the casing and the signature...

Reply:
Yeah, so by convention in React HOC names are starting with the prefix 'with' and with lower case. In this situation, I omit the prefix since it would sound strange ```withFetchWithStatus```

Reply:
I wonder though, why a HOC and not a normal component that just uses `<Fetch>`, `<StatusIndicator>`, and the content (as children)?

Reply:
```
const FetchWithStatus = ({ children }) => {
  return (
    <Fetch>
      {({ err, data }) => {
        return (
          <StatusIndicator>
            {children} // the issue is here - no way to pass the data to the children
          </StatusIndicator>
        )
      }}
    </Fetch>
  )
}
```
Basically there is a way to make it work. If I pass the data to StatusIndicator and StatusIndicator to pass it back to children(data) which is odd since children have access to data already in the parent scope....too much complexity...maybe there is another approach which I cannot see